lexicons/com/atproto/server/checkAccountStatus.json

```diff
@@ -0,0 +1,34 @@
+{
+  "lexicon": 1,
+  "id": "com.atproto.server.checkAccountStatus",
+  "defs": {
+    "main": {
+      "type": "query",
+      "description": "Returns the status of an account, especially as pertaining to import or recovery. Can be called many times over the course of an account migration.",
+      "output": {
+        "encoding": "application/json",
+        "schema": {
+          "type": "object",
+          "required": [
+            "activated",
+            "uploadedRepo",
+            "importedRepo",
+            "importedPrivateState",
+            "expectedBlobs",
+            "importedBlobs"
+          ],
+          "properties": {
+            "activated": { "type": "boolean" },
+            "didMigrated": { "type": "boolean" },
+            "uploadedRepo": { "type": "boolean" },
+            "importedRepo": { "type": "boolean" },
+            "importRepoError": { "type": "string" },
```

Comment:
It could be nice to make room for an error message and a code/name here.

Reply:
I actually had a change of heart and decided against processing the repo async, that way we can just report the error directly in the request

Reply:
hmm! that is a big change (not being async). it does feel like async would be more general and reliable than keeping a long HTTP request open, and enable a nice dynamic progress bar UI.

but we could add that as an alternative endpoint later.